wordpress-mobile/WordPress-iOS
# Blogging Prompts: Add train of avatars to prompt card

Refs #18250 

This PR adds a new view, `AvatarTrainView`, that is dedicated to showing a train of circular images with a slight overlap in-between each image. The new view is also implemented in the prompt card (although it's just mock data).

Some notes:
- The maximum number of avatars displayed is 3.
- The train of avatars row will be hidden if the answer count is less than one.
- I've also addressed the Share button not scaling up properly in accessibility size. In case I forgot! 😅 
- The `%d answers` is just a plain label for now. Once we decided that it's clickable (and we know where it leads to), it should be updated to UIButton instead.

Here are some screenshots:

Description | Screenshot
-|-
Light scheme | ![train_light](https://user-images.githubusercontent.com/1299411/162450865-cd84c5ce-36ba-4764-8f36-b29cae988e89.png)
Dark scheme | ![train_dark](https://user-images.githubusercontent.com/1299411/162450868-03379d46-5e5d-4d65-941c-d924992ef073.png)
RTL language[^1] | ![train_rtl](https://user-images.githubusercontent.com/1299411/162450862-7bac6976-c4fe-4304-aa3d-a2d6530adfe8.png)
Accessibility size | ![train_a11y](https://user-images.githubusercontent.com/1299411/162450847-6487654c-9508-4ebf-a9b4-0fb58ba7a0f8.png)

## To test

Ensure that the Blogging Prompts and My Site Dashboard feature flags are turned on.

- Go to My Site dashboard, Home tab.
- 🔍 Verify that the train of avatars is displayed correctly.
- Go to App Settings, and switch to dark mode.
- 🔍  Verify that the train of avatars is displayed correctly: the borders are redrawn so there are no white borders.
- Increase font size.
- 🔍  Verify that the `%d answers` are scaling up.
  - Note that when using the largest a11y font size, the label may be truncated. I'll revisit this later in the accessibility round.
- 🔍 Also verify that the Share button now scales correctly.
- Run the app using RTL pseudolanguage.
- 🔍 Verify that the train of avatars and the label's positions are reversed correctly.

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is under development.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is under development.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is under development.

PR submission checklist:


[^1]: For the RTL language screenshot, I used different images marked with numbers to indicate that the images are being laid out right-to-left.
